app/assets/javascripts/deploys.js

```diff
@@ -13,13 +16,22 @@ $(function () {
       $form = $("#new_deploy"),
       $submit = $form.find('input[type=submit]');
 
-  // load changeset when switching to it
-  $("#deploy-tabs a[data-type=github]").click(function (e) {
+  // load changeset when switching to it (for deploy#show)
+  $("#deploy-tabs a[data-toggle=tab][href^='#']").click(function (e) {
       e.preventDefault();
-      var tab = $(this);
-      tab.tab("show");
 
+      var $tab = $(this);
+      var $placeholder = $container.find('.changeset-placeholder');
+      $tab.tab("show");
+
+      // abort if already loaded (either static or via previous changeset load)
+      if ($container.find($tab.attr('href')).length) { return; }
```

Comment:
might be able to solve this by doing `.one('click', function)` instead of `.click` above

Reply:
we don't know if the link maps to a static or something not loaded yet, hence we need this even after the first click